File: App/app/src/main/java/com/cinema/ticket_booking/data/repository/StaffRepository.java
package com.cinema.ticket_booking.data.repository;

import com.cinema.ticket_booking.data.model.response.ApiResponse;
import com.cinema.ticket_booking.data.model.response.StaffDashboardStatsResponse;
import com.cinema.ticket_booking.network.ApiService;

import javax.inject.Inject;
import javax.inject.Singleton;

import retrofit2.Call;

@Singleton
public class StaffRepository {

    private final ApiService apiService;

    @Inject
    public StaffRepository(ApiService apiService) {
        this.apiService = apiService;
    }

    public Call<ApiResponse<StaffDashboardStatsResponse>> getDashboardStats() {
        return apiService.getStaffDashboardStats();
    }
}

File: App/app/src/main/java/com/cinema/ticket_booking/data/repository/StaffRepository.kt
package com.cinema.ticket_booking.data.repository

import com.cinema.ticket_booking.data.model.response.ApiResponse
import com.cinema.ticket_booking.data.model.response.CheckInHistoryItemResponse
import com.cinema.ticket_booking.data.model.response.PageResponse
import com.cinema.ticket_booking.data.model.response.StaffDashboardStatsResponse
import com.cinema.ticket_booking.data.model.response.UpcomingShowtimeResponse
import com.cinema.ticket_booking.network.ApiService
import javax.inject.Inject
import javax.inject.Singleton
import retrofit2.Call

@Singleton
class StaffRepository @Inject constructor(
    private val apiService: ApiService
) {

    fun getDashboardStats(): Call<ApiResponse<StaffDashboardStatsResponse>> {
        return apiService.getStaffDashboardStats()
    }

    fun getUpcomingShowtimes(): Call<ApiResponse<List<UpcomingShowtimeResponse>>> {
        return apiService.getUpcomingShowtimes()
    }

    fun getCheckInHistory(filter: String, page: Int, size: Int): Call<ApiResponse<PageResponse<CheckInHistoryItemResponse>>> {
        return apiService.getCheckInHistory(filter, page, size)
    }
}
